Update status in readme (#183)
JuliaDiff/Diffractor.jl: README.md

- Efficient higher order derivatives through nested AD
- Reasonable compile times
- High flexibility (like Zygote)
- Support for forward/reverse/mixed modes
- Fast Jacobians

This is achieved through a combination of innovations:
- A new lowest level interface (∂⃖ the "AD optic functor" or "diffractor"), more suited to higher order AD
- New capabilities in Base Julia (Opaque closures, inference plugins)
- Better integration with ChainRules.jl
- Demand-driven forward-mode AD (Applying transforms to only those IR statements that contribute to relevant outputs of the function being differentiated)

# Current status
## Current Status: Forward-Mode
Currently, forward-mode is the only fully-functional mode and is now shipping in some closed source products.
It is in a position to compete with [ForwardDiff.jl](https://github.com/JuliaDiff/TaylorDiff.jl), and with [TaylorDiff.jl](https://github.com/JuliaDiff/TaylorDiff.jl).
It is not as battle-tested as ForwardDiff.jl, but it has several advantages.
Primarily, as it is not an operator overloading AD, it frees one from the need to relax type-constants and worry about the types of containers.

> Furthermore, Like TaylorDiff.jl, it supports Taylor series based computation of higher order derviatives.

Author: ```suggestion
Furthermore, Like TaylorDiff.jl, it supports Taylor series based computation of higher order derviatives.
It directly and efficiently uses ChainRules.jl's `frules`, no need for a wrapper macro to import them etc.
```

# General Overview

Diffractor is an experimental next-generation, compiler-based AD system for Julia.

Design goals:
- Ultra high performance for both scalar and array code

> - Efficient higher order derivatives though nested AD

Reviewer: ```suggestion
- Efficient higher order derivatives through nested AD
```

- Reasonable compile times
- High flexibility (like Zygote)
- Support for forward/reverse/mixed modes
- Fast Jacobians

This is achieved through a combination of innovations:
- A new lowest level interface (∂⃖ the "AD optic functor" or "diffractor"), more suited to higher order AD
- New capabilities in Base Julia (Opaque closures, inference plugins)
- Better integration with ChainRules.jl

> - Demand-driven forward-mode AD (Forwards-mode AD that knows which outputs are active and only ADs the path to them.)

Reviewer: ```suggestion
- Demand-driven forward-mode AD (Applying transforms to only those IR statements that contribute to relevant outputs of the function being differentiated)
```

# Current status

> ## Current Status: Forwards-Mode

Reviewer: ```suggestion
## Current Status: Forward-Mode
```

> Recently forwards-mode has been a priority, and now forms a core part of some closed source products.

Reviewer: ```suggestion
Currently, forward-mode is the only fully-functional mode and is now shipping in some closed source products.
```

It is in a position to compete with [ForwardDiff.jl](https://github.com/JuliaDiff/TaylorDiff.jl), and with [TaylorDiff.jl](https://github.com/JuliaDiff/TaylorDiff.jl).

> It is certainly no where near as battle-tested as ForwardDiff.jl, but it has several advantages.

Reviewer 2: mention chainrules support

> It is certainly no where near as battle-tested as ForwardDiff.jl, but it has several advantages.

Reviewer: ```suggestion
It is not as battle-tested as ForwardDiff.jl, but it has several advantages.
```

Primarily, as it is not an operator overloading AD, it frees one from the need to relax type-constants and worry about the types of containers.
Furthermore, Like TaylorDiff.jl, it supports Taylor series based computation of higher order derviatives.

> One limitation over ForwardDiff.jl is a lack of chunking support, to pushforward multiple basies at once.

Reviewer: ```suggestion
One limitation over ForwardDiff.jl is a lack of chunking support, to pushforward multiple bases at once.
```

## Current Status: Reverse-Mode

> Reverse mode has recently been deprioritized.
> While originally, a higher priority and working well, it is now a secondary priority.<br>
> ⚠️ **Reverse Mode support should be considered experimental, and may break without warning, and may not be fixed rapidly.** ⚠️ <br>
> 
> With that said, issues and PRs for reverse mode continue to be appreciated.
> 

Reviewer: ```suggestion
Improved reverse mode support is planned for a future release.
While reverse mode was originally implemented and working, it has been stripped out until such a time as it can be properly implemented on top of new Julia compiler changes.<br>
⚠️ **Reverse Mode support should be considered experimental, and may break without warning, and may not be fixed rapidly.** ⚠️ <br>

With that said, issues and PRs for reverse mode continue to be appreciated.
```
